clustering/ejb/WFLY-14359_Create_distributable-ejb_subsystem.adoc

```diff
@@ -0,0 +1,174 @@
+* defining a client mappings registry provider for the EJB client and integrating that provider into the ejb3 subsystem
+
+The remaining two features will be completed in separate RFEs:
+
+** WFLY-7628 will address this for distributed timer management
+** WFLY-15432 will address this for singleton service
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14953
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-7628
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:tborgato@redhat.com[Tommaso Borgato]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [x] QE
+
+=== Affected Projects or Components
+
+Clustering, EJB
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Backwards Compatibility
+
+These changes affect backward compatability (c.f. legacy configuration of cache factories, passivation stores,
+client mappings registries, etc)
+
+=== Default Configuration
+
+Here is an example of the way the new subsystem will look, for the case of specifying a bean management provider:
+----
+<subsystem xmlns="urn:jboss:domain:distributable-ejb:1.0" default-bean-management="infinispan">
+    <infinispan-bean-management name="infinispan" cache-container="ejb" cache="dist" max-size="10"/>
+</subsystem>
+----
+and here is its corresponding integration into the ejb3 subsystem:
+----
+<subsystem xmlns="urn:jboss:domain:ejb3:10.0">
+     ...
+     <caches>
+        <simple-cache name="simple-cache"/>
+        <distributable-cache name="distributable-cache" bean-management="infinispan"/>
+    </caches>
+    ...
+ </subsystem>
+----
+Note that the distributable-cache element takes an optional bean-management attribute, which is a reference to
+the bean management provider in the new subsystem. A similar arrangment will be used for client mappings registries.
+
+As before, for each EJB SFSB, the @Cache annotation (resp. deployment descriptor configuration) may be used to specify
+the name of the desired cache factory defined in the ejb3 subsystem. If no @Cache annotation (resp. deployment descriptor
+configuration) is present, a suitable default cache will be used for the bean.
+
+=== Importing Existing Configuration
+
+=== Deployments
+
+A new deployment configuration file, distributable-ejb.xml, will be introduced to permit specifying default providers
```

Comment:
A new deployment configuration namespace, specified via jboss-all.xml or a separate distributable-ejb.xml, will be introduced to permit specifying default providers ...

Reply:
Done.